internal/commands/image/list.go

```diff
@@ -52,148 +51,157 @@ func listCmd(dockerCli command.Cli) *cobra.Command {
+	id, ok := ref.(store.ID)
 	if !ok {
 		var err error
-		id, err = store.FromBundle(p.bundle)
+		id, err = store.FromBundle(bundle)
 		if err != nil {
 			return "", err
 		}
 	}
 	return stringid.TruncateID(id.String()), nil
 }
 
-func printHeaders(w io.Writer, listColumns []imageListColumn) {
-	var headers []string
-	for _, column := range listColumns {
-		headers = append(headers, column.header)
+func printHeaders(w io.Writer, digests bool) {
+	headers := []string{"REPOSITORY", "TAG"}
+	if digests {
+		headers = append(headers, "DIGEST")
 	}
+	headers = append(headers, "APP IMAGE ID", "APP NAME", "CREATED")
 	fmt.Fprintln(w, strings.Join(headers, "\t"))
 }
 
-func printValues(w io.Writer, ref pkg, listColumns []imageListColumn) {
-	var values []string
-	for _, column := range listColumns {
-		values = append(values, column.value(ref))
+type imageDesc struct {
+	ID         string        `json:"id,omitempty"`
+	Name       string        `json:"name,omitempty"`
+	Repository string        `json:"repository,omitempty"`
+	Tag        string        `json:"tag,omitempty"`
+	Digest     string        `json:"digest,omitempty"`
+	Created    time.Duration `json:"created,omitempty"`
+}
+
+func getImageDesc(bundle *relocated.Bundle, ref reference.Reference) imageDesc {
+	var id string
+	id, _ = getImageID(bundle, ref)
+	var repository string
+	if n, ok := ref.(reference.Named); ok {
+		repository = reference.FamiliarName(n)
+	}
+	var tag string
+	if t, ok := ref.(reference.Tagged); ok {
+		tag = t.Tag()
+	}
+	var digest string
+	if t, ok := ref.(reference.Digested); ok {
+		digest = t.Digest().String()
+	}
+	var created time.Duration
+	if payload, err := packager.CustomPayload(bundle.Bundle); err == nil {
+		if createdPayload, ok := payload.(packager.CustomPayloadCreated); ok {
+			created = time.Now().UTC().Sub(createdPayload.CreatedTime())
+		}
+	}
+	return imageDesc{
+		ID:         id,
+		Name:       bundle.Name,
+		Repository: repository,
+		Tag:        tag,
+		Digest:     digest,
+		Created:    created,
 	}
-	fmt.Fprintln(w, strings.Join(values, "\t"))
 }
 
-func getImageListColumns(options imageListOption) []imageListColumn {
-	columns := []imageListColumn{
-		{"REPOSITORY", func(p pkg) string {
-			if n, ok := p.ref.(reference.Named); ok {
-				return reference.FamiliarName(n)
-			}
-			return "<none>"
-		}},
-		{"TAG", func(p pkg) string {
-			if t, ok := p.ref.(reference.Tagged); ok {
-				return t.Tag()
-			}
-			return "<none>"
-		}},
+func (desc imageDesc) humanDuration() string {
+	if desc.Created > 0 {
+		return units.HumanDuration(desc.Created) + " ago"
 	}
-	if options.digests {
-		columns = append(columns, imageListColumn{"DIGEST", func(p pkg) string {
-			if t, ok := p.ref.(reference.Digested); ok {
-				return t.Digest().String()
-			}
-			return "<none>"
-		}})
+	return ""
+}
+
+func (desc imageDesc) println(w io.Writer, digests bool) {
+	values := []string{}
```

Comment:
Should we have something like this instead https://github.com/docker/cli/pull/2107/files#diff-6c8144102a0cd8ad4191c8c221adc70bL14 ? A default format applied ?

Reply:
I thought about this option, but need to admit I'm pretty bad in writting go templates :)
will look into this